### PR TITLE
Add support for stack projects timings, bump to GHC 9

### PR DESCRIPTION
**Pull request by swamp-agr**
- Bump upper bound for `base`.
- Add source file location to `GhcFile` (disable composing filepath since it's already in place).
- Add initial support of `.stack-work` internals.

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -51,11 +51,22 @@ main = do
             case stripPrefix dir file of
               Nothing -> Left file
               Just x -> case splitDirectories x of
-                ("/" : "build" : hostOs : ghcVersion : packageName : componentType : subComponent : "build" : modulePath) -> Right GhcFile{..}
+                ("/" : "build" : hostOs : ghcVersion : packageName : componentType : subComponent : "build" : modulePath) ->
+                  let srcFilePath = file
```
Comment (qnikst) [inline Main.hs]: let's write `= partitionEithers $ files <&> \srcFilePath ->` in line 50 in order to avoid repetition?

File: src/GhcFile.hs [hunk 2]
```diff
@@ -30,8 +30,7 @@ data GhcFile = GhcFile
 --
 -- It looks terrible, seems a wrong abstraction is here.
 rebuildFilePath :: FilePath -> GhcFile -> FilePath
-rebuildFilePath base GhcFile{..} =
-  base </> "build" </> hostOs </> ghcVersion </> packageName </> componentType </> subComponent </> "build" </> joinPath modulePath
+rebuildFilePath _base GhcFile{..} = srcFilePath
```
Comment (qnikst) [inline src/GhcFile.hs]: Seems that `_base`, let's remove it?

**Review by qnikst — COMMENTED** [review]
Thanks a lot, only a few comments about the code

File: Main.hs (hunk 1 above)
Comment (swamp-agr) [reply Main.hs]: Fixed.

File: src/GhcFile.hs (hunk 2 above)
Comment (swamp-agr) [reply src/GhcFile.hs]: Fixed.

**Review by qnikst — APPROVED** [review]

**Pull request merged**